tests/data/modules/exp/plugins/__init__.py

```diff
@@ -39,3 +40,29 @@ class Test2(resources.PurgeableResource):
     """
 
     fields = ("name", "agent", "mydict", "mylist")
+
+
+@resources.resource("exp::RequiresTest", agent="agent", id_attribute="name")
+class RequiresTest(resources.PurgeableResource):
+
+    fields = ("name", "agent", "do_break")
+
+
+@dependency_manager
+def bad_loops(_, all_resources):
+    my_resources = {}
+
+    for resource in all_resources.values():
+        if resource.id.get_entity_type() == "exp::RequiresTest":
+            my_resources[resource.id] = resource
```

Comment:
Dict comprehension for clarity? Not sure how relevant remarks like these are in test cases.
```suggestion
    my_resources = {
        resource.id: resource for resource in all_resources.values() if resource.id.get_entity_type() == "exp::RequiresTest"
    }
```